client/starwhale/api/_impl/dataset/loader.py

```diff
@@ -92,6 +94,44 @@ def _do_iter_row(self) -> t.Generator[TabularDatasetRow, None, None]:
             for row in self.tabular_dataset.scan():
                 yield row
         else:
+
+            def _travel_link(obj: t.Any) -> t.List[Link]:
+                _lks = []
+                if isinstance(obj, Link):
+                    _lks.append(obj)
+                elif isinstance(obj, dict):
+                    for v in obj.values():
+                        _lks.extend(_travel_link(v))
+                elif isinstance(obj, (list, tuple)):
+                    for v in obj:
+                        _lks.extend(_travel_link(v))
+                elif isinstance(obj, SwObject):
+                    for v in obj.__dict__.values():
+                        _lks.extend(_travel_link(v))
+                return _lks
+
+            def sign_uris(uris: t.List[str]) -> str:
+                r = (
+                    CloudRequestMixed()
+                    .do_http_request(
+                        f"/project/{self.dataset_uri.project}/{self.dataset_uri.object.typ}/{self.dataset_uri.object.name}/version/{self.dataset_uri.object.version}/sign-links",
```

Comment:
why do we use `sign-link` url for single URL or batch URLs in the params?

Reply:
done